src/LogikUI/File/FileManager.cs

```diff
@@ -0,0 +1,337 @@
+{
+    /* FILE STRUCTURE:
+     *  circuit
+     *    wires
+     *      wire
+     *        from < -x: int, y: int
+     *        to < -x: int, y: int
+     *      ...
+     *    components
+     *      component
+     *        type
+     *          "buffer" | "and" | "or" | "xor"
+     *        location < -x: int, y: int
+     *        orientation
+     *          "north" | "east" | "south" | "west"
+     *      ...
+     *    labels
+     *      label < -size: int(10, 100)
+     *        location < -x: int, y: int
+     *        text
+     *          <string>
+     *      ...
+     */
+
+    /// <summary>
+    /// Handles writing and parsing of project files.
+    /// </summary>
+    static class FileManager
+    {
+        private static Dictionary<string, ComponentType> types = new Dictionary<string, ComponentType>()
+        {
+            { "buffer", ComponentType.Buffer },
+            { "and", ComponentType.And },
+            { "or", ComponentType.Or },
+            { "xor", ComponentType.Xor },
+        };
+        private static Dictionary<string, Circuit.Orientation> orientations = new Dictionary<string, Circuit.Orientation>()
+        {
+            { "north", Circuit.Orientation.North },
+            { "south", Circuit.Orientation.South },
+            { "west", Circuit.Orientation.West },
+            { "east", Circuit.Orientation.East },
+        };
+        private static string filename;
+
+        /// <summary>
+        /// True, if FileManager.Open or FileManager.Save hasn't been called.
+        /// </summary>
+        public static bool IsNew { get; private set; } = true;
+        public static List<Wire> Wires { get; set; } = new List<Wire>();
+        public static List<InstanceData> Components { get; set; } = new List<InstanceData>();
+        public static List<TextLabel> Labels { get; set; } = new List<TextLabel>();
+
+        private static void ValidationCallBack(object sender, ValidationEventArgs args)
+        {
+            if (args.Severity == XmlSeverityType.Warning)
+                Output.WriteWarning(args.Message);
+            else
+                throw args.Exception;
+        }
+
+        /// <summary>
+        /// Parses a project file. The results are saved in FileManager.Wires, FileManager.Components, and FileManager.Labels.
+        /// </summary>
+        /// <param name="filename">The path of the project file.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidProjectDataException">The XML-file doesn't match with the schema.</exception>
+        /// <exception cref="ArgumentException">Unable to access the file.</exception>
+        public static void Load(string filename)
+        {
+            Vector2i getPos(XmlNode pos)
+            {
+                int x = int.Parse(pos.Attributes["x"].InnerText);
+                int y = int.Parse(pos.Attributes["y"].InnerText);
+
+                return new Vector2i(x, y);
+            }
```

Comment:
Name should be in pascal case and the local function should be marked as `static` so that is avoids any accidental captures.